sdk/keyvault/azure-security-keyvault-keys/src/delete_key_operation.cpp

```diff
@@ -17,7 +17,8 @@ std::unique_ptr<Azure::Core::Http::RawResponse>
 Azure::Security::KeyVault::Keys::DeleteKeyOperation::PollInternal(
     Azure::Core::Context const& context)
 {
-  std::unique_ptr<Azure::Core::Http::RawResponse> rawResponse;
+  std::unique_ptr<Azure::Core::Http::RawResponse> rawResponse = std::move(m_rawResponse);
+
```

Comment:
I don't yet fully understand how moving the raw response member variable into the local unique ptr fixes the issue, and why there was a null pointer error to begin with.

Both code paths in the try *and* catch below re-assign a value to the `rawResponse` local anyway.

Reply:
Oh... in the case of `if (!isDone())` being false, we are returning back a null unique ptr, since no code path would otherwise assign `rawResponse`.

Reply:
That said, we now seem to be going against what this comment is suggesting, which is to return a copy of the raw response, rather than transferring ownership of it via the `move`:

https://github.com/Azure/azure-sdk-for-cpp/blob/397b8a9d8398212bc12b7bbd7528d854eb32a075/sdk/keyvault/azure-security-keyvault-keys/src/delete_key_operation.cpp#L56-L59